libraries/shared/watcher/storage_watcher.go

```diff
@@ -31,22 +31,18 @@ import (
 	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
 )
 
-type StorageWatcher struct {
-	db             *postgres.DB
-	StorageFetcher fetcher.IStorageFetcher
-	Queue          storage.IStorageQueue
-	Transformers   map[common.Address]transformer.StorageTransformer
+type IStorageWatcher interface {
+	AddTransformers(initializers []transformer.StorageTransformerInitializer)
+	Execute(diffsChan chan utils.StorageDiff, errsChan chan error, queueRecheckInterval time.Duration)
 }
 
-func NewStorageWatcher(fetcher fetcher.IStorageFetcher, db *postgres.DB) StorageWatcher {
-	transformers := make(map[common.Address]transformer.StorageTransformer)
-	queue := storage.NewStorageQueue(db)
-	return StorageWatcher{
-		db:             db,
-		StorageFetcher: fetcher,
-		Queue:          queue,
-		Transformers:   transformers,
-	}
+type StorageWatcher struct {
+	db                        *postgres.DB
+	StorageFetcher            fetcher.IStorageFetcher
+	Queue                     storage.IStorageQueue
+	Transformers              map[common.Address]transformer.StorageTransformer
+	KeccakAddressTransformers map[common.Address]transformer.StorageTransformer // keccak hash of an address => transformer
```

Comment:
Should this be `map[common.Hash]transformer.StorageTransformer?

Reply:
Oh interesting. It is probably more accurate for it to be a `common.Hash` vs an `common.Address`. I was trying to reuse this structure defined [here](https://github.com/vulcanize/vulcanizedb/blob/staging/libraries/shared/storage/utils/row.go#L27) for a diff, but it looks like that may not work out. I think when we convert from a hash to an address, it ends up cutting off the first several characters to make sure that it is the right length for an address, which isn't ideal. I'll have to think on that a bit to see how best to handle it.